src/string.js

```diff
@@ -0,0 +1,249 @@
+/** Extend string class in javascript to support more string manipulation features */
+
+const ExtendString = {
+
+  /*
+  * Checks if the argument contains vowels
+  * @param {string}
+  * @returns {boolean}
+  * @example
+  * // returns true
+  * 'hello'.hasVowels();
+  */
+  hasVowels() {
+    return /[aeiou]/i.test(this);
+  },
+
+  /*
+  * Converts the argument to uppercase characters
+  * @param {string}
+  * @returns {string}
+  * @example
+  * // returns 'HELLO'
+  * 'hello'.toUpper();
+  */
+  toUpper() {
+    const len = this.length;
+    let cap = '';
+    for (let i = 0; i < len; i += 1) {
+      if (this[i].match(/[a-z]/)) {
+        cap += String.fromCharCode(this.charCodeAt(i) - 32);
+      } else {
+        cap += this[i];
+      }
+    }
+    return cap;
+  },
+
+  /*
+  * Converts the argument to lowercase characters
+  * @param {string}
+  * @returns {string}
+  * @example
+  * // returns 'hello'
+  * 'HELLO'.toLower();
+  */
+  toLower() {
+    const len = this.length;
+    let lower = '';
+    for (let i = 0; i < len; i += 1) {
+      if (this[i].match(/[A-Z]/)) {
+        lower += String.fromCharCode(this.charCodeAt(i) + 32);
+      } else {
+        lower += this[i];
+      }
+    }
+    return lower;
+  },
+
+  /*
+  * Converts the  first character of the argument to uppercase
+  * @param {string}
+  * @returns {string}
+  * @example
+  * // returns 'Hello'
+  * 'hello'.ucFirst();
+  */
+  ucFirst() {
+    return (this[0].toUpper() + this.slice(1));
```

Comment:
**Code Quality - Critical**
You want to make use of ES6 string interpolation as stated in our style guide. Use template syntax instead of using `+` to concatenate strings. Like so:

``` javaScript
return `${this[0].toUpper()}${this.slice(1)}`;
```

Take a look at our Airbnb [style guide](https://github.com/airbnb/javascript#es6-template-literals) on template syntax.